src/dataflow/components/dataflow-program.tsx

```diff
@@ -524,12 +524,43 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
 
   private hasValidOutputNodes = () => {
     const { ui } = this.stores;
-    if (!this.hasRelay() && !this.hasDataStorage()) {
+    const hasRelay = this.hasRelay();
+    const hasDataStorage = this.hasDataStorage();
+    let hasValidRelay = false;
+    let hasValidDataStorage = false;
+    if (hasRelay || hasDataStorage) {
+      this.programEditor.nodes.forEach((n: Node) => {
+        if (n.name === "Relay" && n.data.recentValues) {
+          const input = n.inputs.get(Array.from(n.inputs.keys())[0]);
+          const inputNode = input && input.connections[0] && input.connections[0].output.node;
+          const recentVals: any = inputNode && inputNode.data.recentValues;
+          if (recentVals && !isNaN(recentVals[recentVals.length - 1].nodeValue.val) && n.data.relayList !== "none") {
```

Comment:
nit: I prefer `isFinite(x)` to `!isNaN(x)` -- it catches infinities as well as NaNs and the code reads more clearly than the double-negative of "is not not-a-number".

Reply:
Fixed.